src/main/kotlin/no/nav/k9/los/domene/repository/BehandlingProsessEventKlageRepository.kt

```diff
@@ -0,0 +1,160 @@
+        }
+        val modell = objectMapper().readValue(json, K9KlageModell::class.java)
+
+        return K9KlageModell(modell.eventer.sortedBy { it.eventTid }.toMutableList())
+    }
+
+    fun hentMedLås(tx: TransactionalSession, uuid: UUID): K9KlageModell {
+        val json: String? = tx.run(
+            queryOf(
+                "select data from behandling_prosess_events_klage where id = :id for update",
+                mapOf("id" to uuid.toString())
+            )
+                .map { row ->
+                    row.string("data")
+                }.asSingle
+        )
+
+        Databasekall.map.computeIfAbsent(object {}.javaClass.name + object {}.javaClass.enclosingMethod.name) { LongAdder() }
+            .increment()
+        if (json.isNullOrEmpty()) {
+            return K9KlageModell(mutableListOf())
+        }
+        val modell = objectMapper().readValue(json, K9KlageModell::class.java)
+
+        return K9KlageModell(modell.eventer.sortedBy { it.eventTid }.toMutableList())
+    }
+
+    fun fjernDirty(uuid: UUID, tx: TransactionalSession) {
+        tx.run(
+            queryOf(
+                """update behandling_prosess_events_klage set dirty = false where id = :id""",
+                mapOf("id" to uuid.toString())
+            ).asUpdate
+        )
+    }
+
+    fun lagre(uuid: UUID, f: (K9KlageModell?) -> K9KlageModell): K9KlageModell {
+        Databasekall.map.computeIfAbsent(object {}.javaClass.name + object {}.javaClass.enclosingMethod.name) { LongAdder() }
+            .increment()
+        var sortertModell = K9KlageModell(mutableListOf())
+        using(sessionOf(dataSource)) { it ->
+            it.transaction { tx ->
+                val run = tx.run(
+                    queryOf(
+                        "select data from behandling_prosess_events_klage where id = :id for update",
+                        mapOf("id" to uuid.toString())
+                    )
+                        .map { row ->
+                            row.string("data")
+                        }.asSingle
+                )
+
+                val modell = if (!run.isNullOrEmpty()) {
+                    val modell = objectMapper().readValue(run, K9KlageModell::class.java)
+                    f(modell.copy(eventer = modell.eventer.sortedBy { it.eventTid }.toMutableList()))
+                } else {
+                    f(null)
+                }
+                sortertModell =
+                    modell.copy(eventer = (modell.eventer.toSet().toList().sortedBy { it.eventTid }.toMutableList()))
+                val json = objectMapper().writeValueAsString(sortertModell)
+                tx.run(
+                    queryOf(
+                        """
+                    insert into behandling_prosess_events_klage as k (id, data, dirty)
+                    values (:id, :data :: jsonb, true)
+                    on conflict (id) do update
+                    set data = :data :: jsonb, dirty = true
```

Comment:
dirty? 😆

Reply:
flagger at vi har opprettet oppgaveversjoner for den aktuelle json-blobben, og at det ikke har kommet nye eventer inn fra k9-sak siden forrige gang denne raden ble prosessert. Brukes for å slippe å behandle hele tabellen hver gang jobben kjører

Reply:
tenkte bare på `dirty` navnet :D Når jeg hører dirty så tenker jeg noe ikke stemmer helt og må ryddes opp :P Men er med på at det kan være at et event ikke er prosessert.

Reply:
Vanlig begrep rundt synkronisering av data :nerd_face:

Reply:
Det er jo i grunn det det betyr. At noe ikke er "ryddet" i oppgavemodellen